src/Request.hs

```diff
@@ -70,9 +64,14 @@ data ContractInfo = ContractInfo
     } deriving (Show, Generic, ToJSON, FromJSON)
 
 {-# INLINABLE mkValidator #-}
-mkValidator :: ContractInfo -> RequestDatum -> RequestRedeemer -> ScriptContext -> Bool
-mkValidator contractInfo@ContractInfo{..} dat rdm ctx = validate
+mkValidator :: ContractInfo -> RequestDatum -> TokenName -> ScriptContext -> Bool
+mkValidator contractInfo@ContractInfo{..} dat lenderTn ctx = validate
   where
+    getLendDate :: Maybe POSIXTime
+    getLendDate = case ivFrom (U.range ctx) of
+      LowerBound (Finite x) _ -> Just x
```

Comment:
In your use-case it should be the `UpperBound`, as this transaction is created by the lender whose incentive is to start earning interest as soon as possible.

Sorry if I mislead you by an example in the Discord.